core/src/autogluon/core/utils/utils.py

```diff
@@ -412,7 +412,11 @@ def compute_weighted_metric(y, y_pred, metric, weights, weight_evaluation=None,
     try:
         weighted_metric = metric(y, y_pred, sample_weight=weights, **kwargs)
     except (ValueError, TypeError, KeyError):
-        logger.log(30, f"WARNING: eval_metric='{metric.name}' does not support sample weights so they will be ignored in reported metric.")
+        if hasattr(metric, 'name'):
+            metric_name = metric.name
+        else:
+            metric_name = metric
```

Comment:
str(metric)?

Reply:
I don't think it matters in the context I am using it.